### ISSUE TITLE
Allow initialization on minimal environment

### ISSUE THREAD

**Issue by bbockelm** [PelicanPlatform/pelican#226]
v7.1.0 cannot be run as a `stash_plugin` because the config system errors out at `os.UserHomeDir()` when `$HOME` isn't set:

```
$ HOME= ~/stash_plugin -d -infile $PWD/stash_plugin.in -outfile /tmp/stash_plugin.out
Error: $HOME is not defined
```

Since this occurs in the `init()` function, the binary exits before the output file can be created (and has no place to put an error).

There's another error that occurs if `HOME` is defined but not a writable directory (a reasonably common situation). See:

```
$ HOME=/dev/null ~/stash_plugin -d -infile $PWD/stash_plugin.in -outfile /tmp/stash_plugin.out
Error: Failure when creating a directory for the monitoring data: mkdir /dev/null: not a directory
```

This is because `InitServer()` is called in the `init()` function in `origin_serve.go` (meaning that `InitServer` is _always_ called, regardless of what's done by the binary).  Rather, InitServer should probably be in a `PersistentPreRunE` function for the server-related objects initialized by cobra.

**Comment by jhiemstrawisc** [comment]
@joereuss12 Can you confirm that the fixes we merged cover these failure cases in HTCondor?

**Comment by joereuss12** [comment]
Yes, I tested it over on submit1 and it now works as `stash_plugin` without any issues and functioned how it should

**Comment by jhiemstrawisc** [comment]
Closing as fixed by PR #230
